cyctl/cmd/apply.go

```diff
@@ -89,6 +89,44 @@ func applyYaml(yamlFile []byte, config *rest.Config) error {
 		if err != nil {
 			return err
 		}
+
+		if telemetry {
+			if obj.GetKind() == "Deployment" {
+				labels, _, _ := unstructured.NestedMap(obj.Object, "metadata", "labels")
+				if labels != nil {
+					appLabel, _, _ := unstructured.NestedString(labels, "app")
+					if appLabel == "cyclops-ctrl" {
+						containers, _, _ := unstructured.NestedSlice(obj.Object, "spec", "template", "spec", "containers")
+						if len(containers) > 0 {
+							container := containers[0].(map[string]interface{})
+							env, _, _ := unstructured.NestedSlice(container, "env")
+							newEnv := map[string]interface{}{
+								"name":  "DISABLE_TELEMETRY",
+								"value": "true",
+							}
+							env = append(env, newEnv)
+							err := unstructured.SetNestedSlice(container, env, "env")
+							if err != nil {
+								log.Fatal(err)
+							}
+							containers[0] = container
+							err = unstructured.SetNestedSlice(obj.Object, containers, "spec", "template", "spec", "containers")
+							if err != nil {
+								log.Fatal(err)
+							}
+						}
+						err = doServerSideApply(context.TODO(), config, obj)
```

Comment:
Are you first deploying the controller without telemetry and then updating it? I would prefer if just alter the object and then call the `doServerSideApply` with it